From 24c2f76ab5b1c1d2b4bc26f95ed68e44f6768b8b Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Wed, 3 Jun 2015 19:30:59 +0800
Subject: [PATCH] MLK-10994-4 usb: ehci: Keep portsc unchanging when power off
 port

If PORT_RWC_BITS is set, it will still be cleared by writing 1.
portsc.ptw can only be changed when PHY clock is off and portsc.phcd is set,
if not, the PHY will be error, so we keep portsc.ptw unchanging
in this patch.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/host/ehci-hcd.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/ehci-hcd.c b/drivers/usb/host/ehci-hcd.c
index ab5d59e..bfe1adc 100644
--- a/drivers/usb/host/ehci-hcd.c
+++ b/drivers/usb/host/ehci-hcd.c
@@ -331,7 +331,8 @@ static void ehci_turn_off_all_ports(struct ehci_hcd *ehci)
 	int	port = HCS_N_PORTS(ehci->hcs_params);
 
 	while (port--) {
-		ehci_writel(ehci, PORT_RWC_BITS,
+		u32 portsc = ehci_readl(ehci, &ehci->regs->port_status[port]);
+		ehci_writel(ehci, portsc,
 				&ehci->regs->port_status[port]);
 		spin_unlock_irq(&ehci->lock);
 		ehci_port_power(ehci, port, false);
-- 
1.8.0

